### PR TITLE
fix(ci): call chart releaser directly.

### LINKED ISSUE #574: Release jobs are triggered for renovate bot PRs
Example PR: [chore(deps): update actions/checkout action to v4.2.2](https://github.com/kubewarden/helm-charts/pull/571)

Release job: https://github.com/kubewarden/helm-charts/actions/runs/11551186609
E2E tests: https://github.com/kubewarden/helm-charts/actions/runs/11551207278

### PR DESCRIPTION
**Pull request by jvanz**
## Description

Updates the release CI to call Chart releaser directly to allow us more control over the process on how the artifacts are signed and uploaded.

Fix #574

### CODE REVIEW

File: .github/workflows/helm-chart-release.yml [hunk 1]
```diff
@@ -80,13 +81,73 @@ jobs:
...
+            # Reindex the repository
+            cr index --config ${CONFIG_FILE} -o ${OWNER} -r ${REPO} --push --token ${CR_TOKEN} --index-path .
+            echo "Repository indexed!"
+
+            # Publish the charts to the OCI registry and sign them
+            for chart_path in $(find ${PACKAGE_PATH} -maxdepth 1 -mindepth 1 ); do
+              echo "Pushing chart $chart_path to ghcr.io"
+              chart_name=$(helm show chart ${chart_path} | yq ".name")
+              REGISTRY="ghcr.io/$GITHUB_REPOSITORY_OWNER/charts"
+              echo "REGISTRY=${REGISTRY}" >> "$GITHUB_ENV"
```
Comment (flavio) [inline .github/workflows/helm-chart-release.yml:142]: this is done inside of a for loop, we're going to add this line multiple times, only the last occurrence is going to be used. Do we really need to do that?

Comment (jvanz) [reply .github/workflows/helm-chart-release.yml]: Thanks, I've missed that! Fixed! :)

**Review by flavio — APPROVED** [review]
LGTM, how are we going to test it?

**Comment by jvanz** [comment]
> LGTM, how are we going to test it?

I've tested it in my fork. I'll document my tests later in this PR

**Comment by flavio** [comment]
@jvanz ship it! :rocket:

**Pull request merged**